### ISSUE TITLE
Training mosaics show rotated segmentation mask for an image with correct segmentation mask in training data

### ISSUE THREAD

**Issue by nehashinde-canopy** [ultralytics/ultralytics#8548]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello all,

I have trained yolov8 nano for segmentation on dataset: https://universe.roboflow.com/ai-widetech-kpxhm/nws_gdwdvhw_y1/dataset/3/images/?split=train&numImages=40
I have attached a training mosaic file where we can see a rotated segmentation mask for a training sample. Is this due to the augmentation or is this a mistake made by the model? Or am I going wrong somewhere?
![train_batch2](https://github.com/ultralytics/ultralytics/assets/160705335/f4766fca-7cfa-4e3e-a46b-f94cfb3e0126)
I checked my training data, the polygon mask is correctly placed on the training image in training data. I used following script to check the placement of the polygon mask on the training image:

```
import cv2
import numpy as np
from random import randint

with open('/path/to/file/8ed81b84-IMG_20220525_094630_jpg.rf.23e775d557f91c0f84e61a773ffaa92a.txt', 'r') as f:
    labels = f.read().splitlines()
img = cv2.imread('/path/to/file/8ed81b84-IMG_20220525_094630_jpg.rf.23e775d557f91c0f84e61a773ffaa92a.jpg')
h,w = img.shape[:2]

for label in labels:
    class_id, *poly = label.split(' ')
    poly = np.asarray(poly,dtype=np.float16).reshape(-1,2) # Read poly, reshape
    poly *= [w,h] # Unscale
    cv2.polylines(img, [poly.astype('int')], True, (randint(0,255),randint(0,255),randint(0,255)), 2) # Draw Poly Lines
    cv2.fillPoly(img, [poly.astype('int')], (randint(0,255),randint(0,255),randint(0,255)), cv2.LINE_AA) # Draw area
    cv2.imshow('img with poly', img)
    cv2.waitKey(0) 
```

This is my training image and the label file:
![8ed81b84-IMG_20220525_094630_jpg rf 23e775d557f91c0f84e61a773ffaa92a](https://github.com/ultralytics/ultralytics/assets/160705335/403ff172-0bf5-4228-88ab-d75d32bd92a1)

The label file corresponding to above training image:
[8ed81b84-IMG_20220525_094630_jpg.rf.23e775d557f91c0f84e61a773ffaa92a.txt]
(https://github.com/ultralytics/ultralytics/files/14447318/8ed81b84-IMG_20220525_094630_jpg.rf.23e775d557f91c0f84e61a773ffaa92a.txt)

**Comment by nehashinde-canopy** [comment]
Should'nt the segmentation mask be rotated as well when an image is rotated?

**Comment by glenn-jocher** [comment]
@nehashinde-canopy hey there! 👋

Absolutely, when an image is rotated during augmentation, the segmentation mask should also be rotated to match the new orientation of the image. However, based on your augmentation settings, it looks like you've disabled augmentations (`augment=False`), including rotation (`degrees=0`). This means no rotation should be applied to your images or masks during training.

If you're observing rotated masks without corresponding image rotations, it might be worth double-checking your dataset or the specific code handling augmentations. Also, ensure that the version of YOLOv8 you're using is up to date, as improvements and bug fixes are regularly made.

For enabling rotation as an augmentation, you can adjust the `degrees` parameter in your settings. Here's a quick example:

```yaml
augment: True
degrees: 45  # Rotate images and masks up to ±45 degrees
```

If the issue persists, feel free to share more details or examples, and we'll be happy to take a closer look! 😊

**Comment by Shadowfear36** [comment]
I run into a similar issue when trying to train a segmentation model on yoloV8. I know my masks are fine as I've created different versions specifically for a UNet model and had no problems. I noticed that upon training, it says it's applying augmentations that I explicitly set to not be used in my hyperparameters file. I even removed albumentations as a dependency to ensure that wasnt being used either. Something about v8's augmentations does not handle polygonal masks very well. Completely ruins the results of the model.

**My Training Engine Settings:**

```
task=segment, mode=train, model=./yolov8m-seg.pt, data=./seg-stain.yaml, epochs=300, time=None, patience=16, batch=16, imgsz=1952, save=True, save_period=-1, cache=False, device=[1, 2, 3, 4], workers=8, project=sam_full-seg-m, name=v1.05, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=True, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.00851, lrf=0.01068, momentum=0.98, weight_decay=0.00041, warmup_epochs=2.67741, warmup_momentum=0.79506, warmup_bias_lr=0.1, box=7.31451, cls=0.49546, dfl=1.85936, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.01517, hsv_s=0.67396, hsv_v=0.37273, degrees=0.0, translate=0.09574, scale=0.40565, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.58372, mosaic=0.81039, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=sam_full-seg-m/v1.05
Overriding model.yaml nc=80 with nc=6
```

**My HyperParameter File:**

```
lr0: 0.00851
lrf: 0.01068
momentum: 0.98
weight_decay: 0.00041
warmup_epochs: 2.67741
warmup_momentum: 0.79506
box: 7.31451
cls: 0.49546
dfl: 1.85936
hsv_h: 0.01517
hsv_s: 0.67396
hsv_v: 0.37273
degrees: 0.0
translate: 0.0
scale: 0.0
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.0
mosaic: 0.0
mixup: 0.0
copy_paste: 0.0
```

** My config/data file does not contain any augmentations either **

Something somewhere is causing automatic augmentation techniques. I have even set augment=False in my training scripts

**Comment by Shadowfear36** [comment]
I also face an issue where I notice in my train_batch photos, all the masks have a single point that is entirely off from the rest. Why does this occur?
